tests/test_explicit_profile.py

```diff
@@ -407,6 +417,278 @@ def func4(a):
     temp_dpath.delete()
 
 
+@pytest.mark.parametrize('reset_enable_count', [True, False])
+@pytest.mark.parametrize('wrap_class, wrap_module',
+                         [(None, None), (False, True),
+                          (True, False), (True, True)])
+def test_profiler_add_methods(wrap_class, wrap_module, reset_enable_count):
+    """
+    Test the `wrap` argument for the
+    `LineProfiler.add_class()`, `.add_module()`, and
+    `.add_imported_function_or_module()` (added via
+    `line_profiler.autoprofile.autoprofile.
+    _extend_line_profiler_for_profiling_imports()`) methods.
+    """
+    def write(path, code):
+        path.write_text(ub.codeblock(code))
+
+    script = ub.codeblock('''
+        from line_profiler import LineProfiler
+        from line_profiler.autoprofile.autoprofile import (
+            _extend_line_profiler_for_profiling_imports as upgrade_profiler)
+
+        import my_module_1
+        from my_module_2 import Class
+        from my_module_3 import func3
+
+
+        profiler = LineProfiler()
+        upgrade_profiler(profiler)
+        # This dispatches to `.add_module()`
+        profiler.add_imported_function_or_module(my_module_1{})
+        # This dispatches to `.add_class()`
+        profiler.add_imported_function_or_module(Class{})
+        profiler.add_imported_function_or_module(func3)
+
+        if {}:
+            for _ in range(profiler.enable_count):
+                profiler.disable_by_count()
+
+        # `func1()` should only have timing info if `wrap_module`
+        my_module_1.func1()
+        # `method2()` should only have timing info if `wrap_class`
+        Class.method2()
+        # `func3()` is profiled but don't see any timing info because it
+        # isn't wrapped and doesn't auto-`.enable()` before being called
+        func3()
+        profiler.print_stats(details=True, summarize=True)
+                          '''.format(
+        '' if wrap_module is None else f', wrap={wrap_module}',
+        '' if wrap_class is None else f', wrap={wrap_class}',
+        reset_enable_count))
+
+    with enter_tmpdir() as curdir:
+        write(curdir / 'script.py', script)
+        write(curdir / 'my_module_1.py',
+              '''
+        def func1():
+            pass  # Marker: func1
+              ''')
+        write(curdir / 'my_module_2.py',
+              '''
+        class Class:
+            @classmethod
+            def method2(cls):
+                pass  # Marker: method2
+              ''')
+        write(curdir / 'my_module_3.py',
+              '''
+        def func3():
+            pass  # Marker: func3
+              ''')
+        proc = ub.cmd([sys.executable, str(curdir / 'script.py')])
```

Comment:
If we bump the minimum ubelt version to 1.3.5, that does allow for items in the argv list to be given as PathLike objects, which are then converted to strings before being passed to Popen.